KickoffTimerPlugin.cpp

```diff
@@ -94,94 +170,112 @@ void KickoffTimerPlugin::onReset(std::string eventName)
 	pBallHitted.text = "";
 	
 	spawn = getSpawnLocation();
-	if (spawn != 0) {
-		std::ostringstream os;
-		os << spawn->name << " normal Kickoff time: " << to_string_with_precision(spawn->normalTime, 2) << " seconds";
-		std::string msg = os.str();
-		pDefaultTime.text = msg;
-
-		os.str("");
-		os.clear();
-		if(spawn->personalBest > 0) os << "Personal Best: " << to_string_with_precision(spawn->personalBest, 2) << " seconds";
-		msg = os.str();
-		pBest.text = msg;
+
+	if (nullptr == spawn)
+	{
+		cvarManager->log("Can't get spawnLocation!.");
+		return;
 	}
+
+	std::ostringstream os;
+	os << spawn->name << " normal Kickoff time: " <<
+		to_string_with_precision(spawn->normalTime, DECIMAL_PRECISION) << " seconds";
+	auto msg = os.str();
+	pDefaultTime.text = msg;
+
+	os.str("");
+	os.clear();
+	if(spawn->personalBest > 0) os << "Personal Best: " <<
+		to_string_with_precision(spawn->personalBest, DECIMAL_PRECISION) << " seconds";
+	msg = os.str();
+	pBest.text = msg;
 }
 
 void KickoffTimerPlugin::Render(CanvasWrapper canvas)
 {
 	if (!gameWrapper->IsInGame() || popups.empty() || !spawn != 0)
 		return;
 
-	auto screenSize = canvas.GetSize();
-	for(int i = 0; i < popups.size(); i++)
+	const auto screen_size = canvas.GetSize();
+	for(auto i = 0; i < popups.size(); i++)
 	{
 		auto pop = popups.at(i);
 		if (pop->startLocation.X < 0)
 		{
-			pop->startLocation = {(int)(screenSize.X * 0.35), (int)(screenSize.Y * 0.1 + i * 0.035 * screenSize.Y)};
+			pop->startLocation = 
+			{
+				static_cast<int>(screen_size.X * 0.35),
+				static_cast<int>(screen_size.Y * 0.1 + i * 0.035 * screen_size.Y)
+			};
 		}
 
-		Vector2 drawLoc = { pop->startLocation.X, pop->startLocation.Y };
-		canvas.SetPosition(drawLoc);
-		canvas.SetColor(pop->color.R, pop->color.G, pop->color.B, 255);
+		const Vector2 draw_loc = { pop->startLocation.X, pop->startLocation.Y };
+		canvas.SetPosition(draw_loc);
+		canvas.SetColor(pop->color.R, pop->color.G, pop->color.B, 225);
 		canvas.DrawString(pop->text, 3, 3);
 	}
 }
 
 SpawnLocation* KickoffTimerPlugin::getSpawnLocation() {
-	auto location = gameWrapper->GetLocalCar().GetLocation();
-	for(int i = 0; i < spawnLocations.size(); i++)
+	const auto location = gameWrapper->GetLocalCar().GetLocation();
+
+	for (auto& spawn_location : spawnLocations)
 	{
-		auto it = &spawnLocations.at(i);
-		if (location.X == it->location.X && location.Y == it->location.Y) {
+		const auto it = &spawn_location;

```

Comment:
Refactor to just spawn_location? No need for it anymore, or?

Reply:
It think using 'it' still makes sense. Its shorter to write that compared to spawn_location, but I see what you mean :)

Reply:
ok so pls change that

Reply:
I wasn't as clear as I should have been in the last one. It's shorter to write what's there now (using 'it'), so I think we should keep it